runbotics/runbotics-desktop/src/action/microsoft/excel/excel.service.ts

```diff
@@ -2,22 +2,212 @@ import { Injectable } from '@nestjs/common';
+            },
+        });
+    }
+
+    // https://learn.microsoft.com/en-us/graph/api/range-insert?view=graph-rest-1.0&tabs=http
+    public async setRange(address: string, values: string | any[][]) {
+        const request = `/worksheets/${this.session.worksheetIdentifier}/range(address='${address}')`;
+
+        if (!Array.isArray(values)) {
+            values = JSON.parse(values) as any[][];
+        }
+
+        const newRange: WorkbookRangeUpdateBody = {
+            values,
+        };
+
+        return this.microsoftGraphService.patch(this.createWorkbookUrl(request), newRange, {
+            headers: {
+                'workbook-session-id': this.session.workbookSessionInfo.id,
+            },
+        });
+    }
+
+    private createWorkbookUrl(request: string) {
+        switch (this.session.platform) {
+            case SHARE_POINT:
+                return `/sites/${this.session.siteId}/drives/${this.session.driveId}/items/${this.session.fileId}/workbook`.concat(
+                    request
+                );
+            case ONE_DRIVE:
+                return `/me/drive/root:/${this.session.sessionIdentifier}:/workbook`.concat(request);
+            default:
+                throw new Error('Invalid platform');
+        }
+    }
+
+    private getColumnNumber(column: string): number {
+        //Calculates the column number based on the column letter
+        // e.g.A = 1  AA = 27, AB = 28, etc.
+        // it uses the ASCII code of the letter to calculate the number
+        // e.g A = 65, B = 66, C = 67, etc.
+        // so we subtract 64 from the ASCII code to get the column number
+        return (column.length - 1) * 26 + (column.charCodeAt(column.length - 1) - 64);
+    }
+
+    private async getSiteIdByName(name: string): Promise<Site> {
+        const url = `/sites?search=${name}`;
+        return (await this.microsoftGraphService.get(url))['value'][0];
+    }
+
+    private async getDriveIdBySiteAndListName(siteId: string, listName: string) {
+        const url = `/sites/${siteId}/drives/`;
+
+        const data = (await this.microsoftGraphService.get(url))['value'] as Drive[];
+
+        return data.filter(drive => drive.name === listName)[0].id;
+    }
+
+    private async getItemId(siteId: string, driveId: string, path: string) {
+        const url = `/sites/${siteId}/drives/${driveId}/root:/${path}`;
+
+        const data: DriveItem = await this.microsoftGraphService.get(url);
+
+        return data.id;
+    }
 
-    /**
-     * 1. Open file id/path
-     * 2. Close session
-     * 3. Get cell
-     * 4. Get Range
-     * 5. Set Cell
-     * 6. Set Range
-     */
+    private async createSharePointSession(
+        platform: Platform,
+        sessionIdentifier: SessionIdentifier,
+        worksheetIdentifier: WorksheetIdentifier,
+        siteRelativePath?: string,
+        list?: string
+    ): Promise<Session> {
+        const siteId = (await this.getSiteIdByName(siteRelativePath)).id;
```

Comment:
can it be undefined? if so then you should handle `.id` call to prevent unwanted error to be thrown